woocommerce/woocommerce
# Fix saving bug with Woo AI settings page and simplify.

### Changes proposed in this Pull Request:

Refactors the Woo AI Settings class to avoid a bug in which the "tone of voice" select field and "business description" textarea input values are not POSTed if the "Enable AI features in your store" option is disabled when saving the Advanced->Features settings section.

To see where this was initially discovered along with a screencast of how to reproduce it, go to #39535.
The crux of this is that the enqueued JavaScript on the page was disabling the fields, which ultimately meant that they were not included in the POST data when the page was saved. 
The main fix here is that we are no longer disabling those fields but rather hiding them when the "Enable AI features in your store" option is unchecked.

Closes https://github.com/woocommerce/woocommerce/issues/39535

### How to test the changes in this Pull Request:

![Screen Capture on 2023-08-03 at 15-09-55](https://github.com/woocommerce/woocommerce/assets/6723003/f4fa8203-1380-492e-9686-cceac231cd69)

1. Go to WooCommerce->Settings->Advanced->Features and test saving various settings on this page, most notably, testing that your input values are not reset if you disable AI features and save the page.
2. Make sure that you can update other settings on that page and that they persist when saving.
3. Make sure that the "tone of voice" and "business description" fields are hidden when the enable checkbox is unchecked.
4. Make sure that with the enable checkbox unchecked and saved, if you refresh the page, the "tone of voice" select field and "business description" fields should still be disabled.


### Changelog entry




<details>

#### Significance




#### Type




#### Message Fix Woo AI settings page fields persistence bug when disabling the feature.

#### Comment 

</details>
